src/Stackage/Package/IndexConduit.hs

```diff
@@ -48,30 +44,80 @@ import Distribution.PackageDescription.Parse
 import Distribution.Text (disp, parse)
 import Distribution.Version (VersionRange)
 import qualified Distribution.Text
-import System.FilePath ((</>), (<.>))
-import System.IO (IOMode(ReadMode), hClose, openBinaryFile)
+import Network.HTTP.Client.Conduit
 import Text.PrettyPrint (render)
 
-sourceTarFile
-  :: MonadResource m
-  => Bool -> FilePath -> Producer m Tar.Entry
-sourceTarFile toUngzip fp = do
-  bracketP (openBinaryFile fp ReadMode) hClose $
-    \h -> do
-      lbs <- liftIO $ L.hGetContents h
-      sourceEntries $ Tar.read $ ungzip' lbs
-  where
-    ungzip'
-      | toUngzip = decompress
-      | otherwise = id
+#if MIN_VERSION_http_conduit(2,2,1)
+import Network.HTTP.Simple (httpSource)
+#else
+import qualified Network.HTTP.Client as H
+import qualified Network.HTTP.Client.TLS as H
+
+-- | For future compatibility, copied here from http-conduit-2.2.1
```

Comment:
I'd just include a lower bound and make sure we have a snapshot that has the right package version.